devicetypes/smartthings/tint-remote-controller.src/tint-remote-controller.groovy

```diff
@@ -0,0 +1,310 @@
+				attributeState "level", action:"switch level.setLevel"
+			}
+			tileAttribute ("device.color", key: "COLOR_CONTROL") {
+				attributeState "color", action:"color control.setColor"
+			}
+		}
+		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range:"(2700..6500)") {
+			state "colorTemperature", action:"color temperature.setColorTemperature"
+		}
+
+		main(["switch"])
+		details(["switch", "colorTempSliderControl"])
+	}
+}
+
+private getGROUPS_CLUSTER() { 0x0004 }
+private getADD_GROUP_COMMAND() { 0x00 }
+private getMOVE_TO_COLOR_COMMAND() { 0x07 }
+private getON_OFF_ATTRIBUTE() { 0x0000 }
+private getMOVE_TO_COLOR_TEMPERATURE_COMMAND() { 0x0A }
+private getSTEP_COMMAND() { 0x02 }
+private getWRITE_ATTRIBUTES_COMMAND() { 0x02 }
+
+def parse(String description) {
+    def event = zigbee.getEvent(description)
+	def result = []
+    if (event) {
+        log.debug "Creating event: ${event}"
+       result = createEvent(event)
+    } else {
+        if (isAttrOrCmdMessage(description)) {
+            result = parseAttrCmdMessage(description)
+        } else {
+            log.warn "Unhandled message came in"
+        }
+    }
+	log.debug "Parse returned: ${result}"
+	return result
+}
+
+private parseAttrCmdMessage(description) {
+    def descMap = zigbee.parseDescriptionAsMap(description)
+    if (descMap.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.attrInt == BATTERY_VOLTAGE_ATTRIBUTE) {
+        getBatteryEvent(zigbee.convertHexToInt(descMap.value))
+    } else if (descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
+        getOnOffEvent()
+    } else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_COMMAND) {
+        getColorEvent(descMap)
+    } else if (descMap.clusterInt == zigbee.COLOR_CONTROL_CLUSTER && descMap.commandInt == MOVE_TO_COLOR_TEMPERATURE_COMMAND) {
+		getColorTemperatureEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER && descMap.commandInt == STEP_COMMAND) {
+		getLevelEvent(descMap)
+	} else if (descMap.clusterInt == zigbee.BASIC_CLUSTER && descMap.commandInt == WRITE_ATTRIBUTES_COMMAND) {
+		getButtonEvent(descMap)
+	} else {
+		log.debug "Unhandled message came in, its description map: ${descMap}"
+	}
+}
+
+private isAttrOrCmdMessage(description) {
+    (description?.startsWith("catchall:")) || (description?.startsWith("read attr -"))
+}
+
+private getOnOffEvent() {
+    state.onOffValue = (state.onOffValue == "off" ? "on" : "off")
+    createEvent(name: "switch", value: state.onOffValue)
+}
+
+private getColorEvent(descMap) {
+    def CurrentX = Integer.parseInt(descMap.data[1] + descMap.data[0], 16)
+    def CurrentY = Integer.parseInt(descMap.data[3] + descMap.data[2], 16)
+    def HS = color_xyYtoHSV(CurrentX, CurrentY)
+	return [
+		createEvent(name: "hue", value: HS.H * 100),
+		createEvent(name: "saturation", value: 100)
```

Comment:
The above conversion function also includes the saturation value, shouldn't you use that value instead of always 100?

Reply:
This conversion is returning 100 saturation _almost_ always, anyway. It has some issues with mismatched colors around blue/purple, so to keep consistence between colors on remote and reports (I'm assuming some device issue here, cause same mismatch occurs when I pair remote to Tint's bulb), I've decided to hard-code saturation. However, I've also thought it won't hurt to keep whole conversion for future reference.

Reply:
I left a big comment below in regard to the xyY stuff and it also pertains to this thread.